solver_spec.rb

```diff
@@ -0,0 +1,62 @@
+require_relative 'solver'
+
+describe Solver do
+  solver = Solver.new
+  context 'Getting factorial of a number (n):' do
+    it 'Raise an exception for factorial of negative numbers ie all numbers less than 0' do
+      expect(solver.factorial(-1)).to eq 'Number is less than 0'
+    end
+
+    it 'Return the factorial of 0 as 1' do
+      expect(solver.factorial(0)).to eq 1
+    end
+
+    it 'Return the factorial of 1 as 1' do
+      expect(solver.factorial(1)).to eq 1
+    end
+
+    it 'Return the factorial of numbers greater than 1' do
+      expect(solver.factorial(9)).to eq 362_880
+    end
+
+    it 'Return the factorial of numbers greater than 100' do
+      expect(solver.factorial(101)).to eq Math.gamma(101 + 1)
+    end
+  end
+
+  context 'Reverse a string:' do
+    # it 'Should return a string value' do
+    #   return_value = solver.fizzbuzz(4)
+    #   expect(return_value).to eq 4.to_string
+    # end
+    it 'Should not get an error for empty string' do
+      expect(solver.reverse('')).to eq ''
+    end
+
+    it 'Should reverse a string of one word' do
+      expect(solver.reverse('hello')).to eq 'olleh'
+    end
+
+    it 'Should reverse a string of one word or more' do
+      expect(solver.reverse('hello world!')).to eq '!dlrow olleh'
+    end
+  end
```

Comment:
As suggested by the previous code reviewer it would be nice if you corrected this you test cases are failing, Please correct this and kindly check project requirements again below are the results I'm getting when i try and run your tests.

Your test results:

![test1](https://user-images.githubusercontent.com/79506105/202876163-93f2955d-6735-49d6-9405-5f9a7ee4ec38.PNG)
![Capture2](https://user-images.githubusercontent.com/79506105/202876171-d514ea6a-7891-4e63-b1ec-aa6e08e2b356.PNG)

Reply:
everything is working fine